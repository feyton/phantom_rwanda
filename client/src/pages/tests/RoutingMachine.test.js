import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { MapContainer } from 'react-leaflet';
import RoutingMachine from '../RoutingMachine.js';

describe('RoutingMachine', () => {
  it('should render the RoutingMachine component', () => {
    const elem = renderer
      .create(
        <MapContainer>
          <MemoryRouter>
            <RoutingMachine />)
          </MemoryRouter>
        </MapContainer>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
  it('should test instance of the routing machine', () => {
    const leafletInstance = null;
    const wrapper = mount(
      <MapContainer>
        <RoutingMachine />
      </MapContainer>
    );
    expect(wrapper.instance).not.toEqual(leafletInstance);
  });
});
