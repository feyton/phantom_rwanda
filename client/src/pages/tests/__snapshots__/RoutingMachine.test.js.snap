// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoutingMachine should render the RoutingMachine component 1`] = `<div />`;
